google-cloud-firestore/lib/google/cloud/firestore/query.rb

```diff
@@ -1001,6 +1008,83 @@ def get read_time: nil
         end
         alias run get
 
+        ##
+        # Retrieves the query explanation for the query.
+        # By default the query is only planned, not executed. returning only metrics from the
+        # planning stages. If `analyze` is set to `true` the query will be planned and executed,
+        # returning the full query results alongside both planning and execution stage metrics.
+        #
+        # @param [Time] read_time Reads documents as they were at the given time.
+        #   This may not be older than 270 seconds. Optional
+        #
+        # @param [Boolean] analyze
+        #   Whether to execute the query and return the execution stage metrics
+        #     in addition to planning metrics.
+        #   If set to `false` the query will be planned only and will return planning
+        #      stage metrics without results.
+        #   If set to `true` the query will be executed, and will return the query results,
+        #     planning stage metrics, and execution stage metrics.
+        #   Defaults to `false`.
+        #
+        # Unlike the Enumerator object that is returned from the `Query#get`,
+        # iterating over QueryExplainResult multiple times will not result in
+        # multiple requests to the server. The first set of results will be saved
+        # and re-used instead.
+        #
+        # @example Iterating over results multiple times
+        #   require "google/cloud/firestore"
+        #
+        #   firestore = Google::Cloud::Firestore.new
+        #   query = firestore.col(:cities).where(:population, :>, 100000)
+        #   explanation_result = query.explain analyze: true
+        #   results = explanation_result.to_a
+        #   results_2 = explanation_result.to_a # same results, no re-query
+        #
+        # This is to avoid the situations where the metrics change unpredictably when results are looked at.
```

Comment:
I believe sections delimited by tags like `@param`, `@return`, and `@example` get reordered and combined in specific locations in the documentation. So don't interleave normal prose and tagged sections and expect them to appear in that order. To reduce confusion, you should reorder this to group all normal prose together, all `@param` and `@return` and related tags together, and all `@example` tags together.

Reply:
Done

Reply:
@viacheslav-rostovtsev I don't see any change here?